modules/react-native-wallet/android/src/main/java/com/expensify/wallet/GooglePushProvisioningModule.java

```diff
@@ -0,0 +1,281 @@
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.common.api.ApiException;
+import android.util.Log;
+import org.json.JSONObject;
+import java.nio.charset.Charset;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class GooglePushProvisioningModule extends ReactContextBaseJavaModule {
+  private static final String MODULE_NAME = "GooglePushProvisioning";
+
+  private static final int REQUEST_CODE_PUSH_TOKENIZE = 3;
+  private static final int REQUEST_CREATE_WALLET = 4;
+
+  public static final String ERROR_GET_TOKEN_STATUS = "GET_TOKEN_STATUS_ERROR";
+  public static final String ERROR_GET_TOKEN_STATUS_FAILURE = "GET_TOKEN_STATUS_FAILURE";
+  public static final String ERROR_GET_TOKEN_STATUS_CANCELED = "GET_TOKEN_STATUS_CANCELED";
+  public static final String ERROR_GET_ACTIVE_WALLET_ID = "GET_ACTIVE_WALLET_ID_ERROR";
+  public static final String ERROR_GET_ACTIVE_WALLET_ID_FAILURE = "GET_ACTIVE_WALLET_ID_FAILURE";
+  public static final String ERROR_GET_ACTIVE_WALLET_ID_CANCELED = "GET_ACTIVE_WALLET_ID_CANCELED";
+  public static final String ERROR_GET_STABLE_HARDWARE_ID = "GET_STABLE_HARDWARE_ID_ERROR";
+  public static final String ERROR_GET_STABLE_HARDWARE_ID_FAILURE = "GET_STABLE_HARDWARE_ID_FAILURE";
+  public static final String ERROR_GET_STABLE_HARDWARE_ID_CANCELED = "GET_STABLE_HARDWARE_ID_CANCELED";
+  public static final String ERROR_GET_ENVIRONMENT = "GET_ENVIRONMENT_ERROR";
+  public static final String ERROR_GET_ENVIRONMENT_FAILURE = "GET_ENVIRONMENT_FAILURE";
+  public static final String ERROR_GET_ENVIRONMENT_CANCELED = "GET_ENVIRONMENT_CANCELED";
+  public static final String ERROR_PUSH_PROVISION = "PUSH_PROVISION_ERROR";
+  public static final String ERROR_PUSH_TOKENIZE_CANCELED = "E_PUSH_TOKENIZE_CANCELED";
+  public static final String ERROR_PUSH_TOKENIZE_FAILED = "E_PUSH_TOKENIZE_FAILED";
+
+  private static final String TSP_VISA = "VISA";
+  private static final String TSP_MASTERCARD = "MASTERCARD";
+
+  private Promise promise;
+
+  private TapAndPayClient tapAndPayClient;
+
+  private static final Map<String, Integer> TOKEN_PROVIDER_MAP = new HashMap<String, Integer>() {{
+    put(TSP_VISA, TapAndPay.TOKEN_PROVIDER_VISA);
+    put(TSP_MASTERCARD, TapAndPay.TOKEN_PROVIDER_MASTERCARD);
+    // Add more if necessary
+  }};
+
+  private static final Map<String, Integer> CARD_NETWORK_MAP = new HashMap<String, Integer>() {{
+    put(TSP_VISA, TapAndPay.CARD_NETWORK_VISA);
+    put(TSP_MASTERCARD, TapAndPay.CARD_NETWORK_MASTERCARD);
+    // Add more if necessary
+  }};
+
+
+  public GooglePushProvisioningModule(ReactApplicationContext reactContext) {
+    super(reactContext);
+    reactContext.addActivityEventListener(mActivityEventListener);
+  }
+
+  @Override
+  public void initialize() {
+    super.initialize();
+    tapAndPayClient = TapAndPay.getClient(this.getCurrentActivity());
```

Comment:
When you run the app, the `Null context is not permitted` error is displayed. It's connected to this line; probably it's being initialized too early